### PR TITLE
feat(action): add create new action for compatibility testing

### PR DESCRIPTION
**Pull request by DexterYan**
- add a new action to create a cmx cluster for compatibility testing
- add helm-extra-repos input
```
      Example:
      helm-extra-repos: |
        - repo_name: "cnpg"
          url: "https://cloudnative-pg.github.io/charts"
          namespace: "cnpg-system"
          chart_name: "cloudnative-pg"
        - repo_name: "minio-operator"
          url: "https://operator.min.io"
          namespace: "minio-operator"
          chart_name: "operator"
```

an example to use this action

```
      - name: Create Compatibility Testing
        id: create-compatibility-testing
        uses: replicated-collab/replicated-actions/prepare-cluster@main
        with:
          app-slug: ${{ secrets.REPLICATED_APP }}
          api-token: ${{ secrets.REPLICATED_API_TOKEN }}
          chart: ${{env.chart_name}}-${{ github.event.inputs.version }}.tgz
          kubernetes-distribution: ${{ matrix.cluster.distribution }}
          kubernetes-version: ${{ matrix.cluster.version }}
          helm-chart-name: ${{env.chart_name}}
          namespace: test-${{env.chart_name}}
          helm-run-preflights: false
          helm-extra-repos: |
            - repo_name: "cnpg"
              url: "https://cloudnative-pg.github.io/charts"
              namespace: "cnpg-system"
              chart_name: "cloudnative-pg"
            - repo_name: "minio-operator"
              url: "https://operator.min.io"
              namespace: "minio-operator"
              chart_name: "operator"
```

### CODE REVIEW

**Comment by jdewinne** [comment]
Is this different from the cluster prepare action? Is it the case that the only difference is the `helm-extra-repos`. If that is needed (not sure), maybe we can improve cluster-prepare?

**Comment by DexterYan** [comment]
> Is this different from the cluster prepare action? Is it the case that the only difference is the `helm-extra-repos`. If that is needed (not sure), maybe we can improve cluster-prepare?

That's a good idea. Let me add it to cluster-prepare

**Comment by DexterYan** [comment]
Hey @jdewinne I have updated the PR to keep it in cluster-prepare

File: prepare-cluster/action.yml [hunk 1]
```diff
@@ -156,6 +169,27 @@ runs:
       INPUT_IP-FAMILY: ${{ inputs.ip-family || '' }}
       INPUT_KUBECONFIG-PATH: ${{ inputs.kubeconfig-path || '' }}
       INPUT_EXPORT-KUBECONFIG: ${{ inputs.export-kubeconfig || false }}
+  - uses: azure/k8s-set-context@v4
+    with:
```
Comment (jdewinne) [inline prepare-cluster/action.yml:173]: Should this also include
```
if: ${{ inputs.helm-extra-repos != '' }}
```

Comment (DexterYan) [reply prepare-cluster/action.yml:174]: good catch. I have added it

**Review by jdewinne — APPROVED** [review]

**Pull request merged**